sync-core/src/main/java/com/cloudant/sync/query/QuerySqlTranslator.java

```diff
@@ -423,36 +424,65 @@ protected static SqlParts whereSqlForAndClause(List<Object> clause, String index
                     // since this clause is negated we need to negate the bool value
                     whereClauses.add(convertExistsToSqlClauseForFieldName(fieldName, exists));
                 } else {
+                    String whereClause;
                     String sqlOperator = operatorMap.get(operator);
                     String tableName = IndexManager.tableNameForIndex(indexName);
-                    String whereClause = whereClauseForNot(fieldName, sqlOperator, tableName);
-
-                    whereClauses.add(whereClause);
-                    predicateValue = negatedPredicate.get(operator);
-                    if (validatePredicateValue(predicateValue)) {
-                        sqlParameters.add(String.valueOf(predicateValue));
+                    String argument;
+                    if (operator.equals(IN)) {
+                        // The predicate map value must be a List here.
+                        // This was validated during normalization.
+                        List<Object> inList = (List<Object>) negatedPredicate.get(operator);
+                        argument = buildWhereClauseArgument(inList, sqlParameters);
+                        // If the argument is a single "?" placeholder then we are
+                        // dealing with a single equality comparison.  Therefore,
+                        // we set the operator to $eq.
+                        if (argument != null && argument.equals("?")) {
+                            sqlOperator = operatorMap.get(EQ);
+                        }
```

Comment:
I'm not sure this special casing for the single case is worthwhile, and it seems a bit fragile to me.